client/src/api.js

```diff
@@ -0,0 +1,63 @@
+import axios from 'axios';
+import Cookies from 'js-cookie';
+
+// request interceptor to add the auth token header to requests
+axios.interceptors.request.use(
+	(config) => {
+		if (Cookies.get('accessToken')) {
+			config.headers.Authorization = `Bearer ${Cookies.get('accessToken')}`;
+		}
+		return config;
+	},
+	(error) => {
+		Promise.reject(error);
+	}
+);
+// response interceptor to refresh token on receiving token expired error
+axios.interceptors.response.use(
+	(response) => {
+		return response;
+	},
+	function (error) {
+		const originalRequest = error.config;
+		if (error.response.status === 401 && !originalRequest._retry) {
+			originalRequest._retry = true;
+			return axios
+				.get('api/v1/token')
+				.then((res) => {
+					if (res.status === 200) {
+						return axios(originalRequest);
+					}
+				});
+		}
+		return Promise.reject(error);
```

Comment:
I like that you've changed to axios and now this code will be only in one place. But still, it takes 3 sequential calls to get info.